pkgs/by-name/na/namida/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  git,
+  fetchgit,
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "namida";
+  version = "0.5";
+
+  # Package uses git history to include versioning in binary
+  src = fetchgit {
+    url = "https://github.com/meew0/namida.git";
+    rev = "782ab6baf679f830c4242bf071d8e85743fe77f7";
+    hash = "sha256-SVFlh603D7mFoN+N4M6v3TXRQWNOV5jMPefmz+qOMQU=";
+    leaveDotGit = true;
+    branchName = "namida";
+  };
+  nativeBuildInputs = [ git ];
```

Comment:
This should likely be addressed upstream so we can use `fetchFromGitHub` instead. The package is also non-deterministic since its outputs differ [based on time of compilation](https://github.com/meew0/namida/blob/782ab6baf679f830c4242bf071d8e85743fe77f7/build.rs#L12-L16), which needs to be addressed before it can be added to nixpkgs.

Reply:
I have written a patch which checks whether the package is within a Nix/OS environment at build time.  If so, it does not check git for a revision or generate a build time.  Instead these values come from `env` in the `package.nix` file.  I will be submitting a pull request upstream.  This solves both the non-determinism and the need for `leaveDotGit`.